tests/gis_tests/distapp/tests.py

```diff
@@ -207,12 +211,18 @@ def test_distance_lookups_with_expression_rhs(self):
             self.assertEqual(self.get_names(qs), ['Canberra', 'Hobart', 'Melbourne'])
 
         # With a complex geometry expression
-        self.assertFalse(SouthTexasCity.objects.filter(point__distance_gt=(Intersection('point', 'point'), 0)))
+        self.assertFalse(SouthTexasCity.objects.filter(point__distance_gt=(Union('point', 'point'), 0)))
```

Comment:
This change is needed because NULL is returned for self-intersection on MySQL < 5.7.